Description: Check MALIIT_SERVER_ADDRESS evn to see if we should use that address
 maliit-framework (0.99.1+git20151118+62bd54b-0ubuntu4) vivid; urgency=medium
 .
   * Add a fix that has been merged upstream. Checking the
     MALIIT_SERVER_ADDRESS for the qt input context.
Author: Brandon Schaefer <brandontschaefer@gmail.com>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: https://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- maliit-framework-0.99.1+git20151118+62bd54b.orig/input-context/minputcontext.cpp
+++ maliit-framework-0.99.1+git20151118+62bd54b/input-context/minputcontext.cpp
@@ -60,7 +60,15 @@ MInputContext::MInputContext()
         debug = true;
     }
 
-    QSharedPointer<Maliit::InputContext::DBus::Address> address(new Maliit::InputContext::DBus::DynamicAddress);
+    QSharedPointer<Maliit::InputContext::DBus::Address> address;
+
+    QByteArray maliitServerAddress = qgetenv("MALIIT_SERVER_ADDRESS");
+    if (!maliitServerAddress.isEmpty()) {
+        address.reset(new Maliit::InputContext::DBus::FixedAddress(maliitServerAddress.constData()));
+    } else {
+        address.reset(new Maliit::InputContext::DBus::DynamicAddress);
+    }
+
     imServer = new DBusServerConnection(address);
 
     sipHideTimer.setSingleShot(true);
